wandb/apis/importers/wandb.py

```diff
@@ -0,0 +1,1748 @@
+"""Tooling for the W&B Importer."""
+import filecmp
+import itertools
+import json
+import logging
+import numbers
+import os
+import re
+import shutil
+from dataclasses import dataclass, field
+from datetime import datetime as dt
+from pathlib import Path
+from typing import Any, Dict, Iterable, Iterator, List, Optional, Tuple
+from unittest.mock import patch
+
+import filelock
+import polars as pl
+import requests
+import urllib3
+import yaml
+from wandb_gql import gql
+
+import wandb
+import wandb.apis.reports as wr
+from wandb.apis.public import ArtifactCollection, Run
+from wandb.apis.public.files import File
+from wandb.apis.reports import Report
+from wandb.util import coalesce, remove_keys_with_none_values
+
+from .internals import internal
+from .internals.internal import _thread_local_settings
+from .internals.util import Namespace, for_each
+
+Artifact = wandb.Artifact
+Api = wandb.Api
+Project = wandb.apis.public.Project
+
+ARTIFACT_ERRORS_FNAME = "artifact_errors.jsonl"
+ARTIFACT_SUCCESSES_FNAME = "artifact_successes.jsonl"
+RUN_ERRORS_FNAME = "run_errors.jsonl"
+RUN_SUCCESSES_FNAME = "run_successes.jsonl"
+
+ART_SEQUENCE_DUMMY_PLACEHOLDER = "__ART_SEQUENCE_DUMMY_PLACEHOLDER__"
+RUN_DUMMY_PLACEHOLDER = "__RUN_DUMMY_PLACEHOLDER__"
+ART_DUMMY_PLACEHOLDER_PATH = "__importer_temp__"
+ART_DUMMY_PLACEHOLDER_TYPE = "__temp__"
+
+logger = logging.getLogger("import_logger")
+# target_size = 80 * 1024**3  # 80GB
+
+os.environ["WANDB_IMPORTER_MODE_ENABLED"] = "true"
```

Comment:
So now if a user does `from wandb.apis.importers.wandb import WandbImporter`, and for some reason wants to log a regular run afterwards, the media stuff will stop working again... Is there any way to do some sort of a context manager instead?

Reply:
Yeah we can set/unset the env inside the import function

Reply:
How about something like this?
https://github.com/wandb/wandb/pull/6897/commits/83d985afd525224623dc794b00e5de0c9a293d18#diff-b04756399c59922236afaa7024b2a5260ad64d4ad0ebf305622257486239f93cR891-R937